### ISSUE TITLE
[no-unused-modules] Imports overriding reexports

### ISSUE THREAD

**Issue by Ephem** [benmosher/eslint-plugin-import#1722]
I noticed that `no-unused-modules` with unusedExports trips up on this syntax:

a.js:
```js
export function a() {}

// The following line will wrongly error as unused:
export function aa() {}
```

b.js:
```js
import { a } from './a';

export { aa } from './a';
```

I've tracked this down to [this line](https://github.com/benmosher/eslint-plugin-import/blob/master/src/rules/no-unused-modules.js#L199) in the src, which overwrites any previously detected reexports. The proposed fix is to do something like what is already being done in [these lines](https://github.com/benmosher/eslint-plugin-import/blob/master/src/rules/no-unused-modules.js#L187-L191) to keep both reexported entries and imports.

I can give a shot at throwing together a PR if this looks reasonable?

**Comment by golopot** [comment]
The example is working as intended. Do you mean 
```diff
- export { aa } from './b';
+ export { aa } from './a';
```
?

**Comment by Ephem** [comment]
Ah, yes, sorry for the typo. I updated the first post to be correct. 😄 Thanks for noticing!

**Comment by ljharb** [comment]
Yes, that seems like a good fix - the code was probably assuming that duplicate imports were always merged.

**Comment by Ephem** [comment]
Yeah, having a import and a reexport from the same file isn't a super common pattern. 😄

I have some code that works, just need to write a test and I'll send a PR within the next few days.
